source/slang/slang.cpp

```diff
@@ -808,18 +808,40 @@ SLANG_API char const* spGetTranslationUnitSource(
     int                     translationUnitIndex)
 {
     auto req = REQ(request);
-    return req->translationUnits[translationUnitIndex]->result.outputSource.Buffer();
+    return (char const*)req->translationUnits[translationUnitIndex]->result.outputSource.Buffer();
+}
+
+SLANG_API void const* spGetTranslationUnitCode(
+    SlangCompileRequest*    request,
+    int                     translationUnitIndex,
+    size_t*                 outSize)
+{
+    auto req = REQ(request);
+    Slang::TranslationUnitResult& result = req->translationUnits[translationUnitIndex]->result;
+    *outSize = (size_t)result.outputSource.Count();
```

Comment:
Just for robustness, maybe check if `outSize` is null before writing to it. That lets people use `spGetTranslationUnitCode(req, idx, nullptr)` as a synonym for `spGetTranslationUnitSource(req, idx)`.